head.go

```diff
@@ -520,14 +627,41 @@ func (a *headAppender) AddFast(ref uint64, t int64, v float64) error {
 	return nil
 }
 
+func (a *headAppender) log() error {
+	if a.head.wal == nil {
+		return nil
+	}
+
+	buf := a.head.getBytesBuffer()
+	defer func() { a.head.putBytesBuffer(buf) }()
+
+	var rec []byte
+	var enc RecordEncoder
+
+	if len(a.series) > 0 {
+		rec = enc.Series(a.series, buf)
```

Comment:
Mh, fair point – worth thinking about further. However I'd imagine rollbacks to be extremely rare. Combined with the small fraction of commits that actually create series, I wonder whether this can be an indicator for common issues.

Reply:
Do we agree that this observation has nothing to do with this PR in particular and should be investigated/addressed seperately?

Reply:
This PR is changing key code around this, and will likely affect the design of the fix. I'd prefer to see it in here - should be a matter of writing out series on a rollback too.

Reply:
This is a big PR already. It doesn't change anything about the issue at hand, so I don't see why one would throw this on top here as well.
We can do another PR right after.